### ISSUE TITLE
[possible bug] Page Editing strange behaviour

### ISSUE THREAD

**Issue by bsusala** [xwp/wp-customize-posts#129]
Tried twice, think is consistently behaving:
- open Page1
- open Editor on Page 1
- leave Editor window open
- change to another Page2 in Preview, using actual links
- in Customizer, Back
- now both pages are listed
- click on the other Page2
- Editor window still open, with text from Page1
- make any edit, Save
- Content from Page1 saves to Page2
Not good :)

**Comment by westonruter** [comment]
It's hard to follow from the steps you listed. Could you create a screencast?

**Comment by bsusala** [comment]
Tried again on a live, stripped down test site, the above behaviour does not replicate. Tried again on my local dev, the above reproduced until changing page inside Preview. TWO pages listed for editing, but Edit window retracts and when clicking for edit the second one, it loads the correct #2 text. Very strange. Cannot publicly post a screencast, if you want I will send a PM, no problem.

**Comment by westonruter** [comment]
Are you on WordPress Slack? You can DM me a link to your screencast there.

**Comment by westonruter** [comment]
@bsusala Thanks for the screencast and additional information on Slack. You noted in Slack:

> the "bug" is that if I do not save, there are two pages available for editing, which is more or less a UX "bug". not so critical, after all

And from your screencast the issue as I see it there is:

* You navigate to a page in the Customizer preview
* You open the “Pages” panel and see the one page there listed as expected.
* You open this page section to edit the page.
* You navigate to another page in the preview.
* The previous page's Customizer section is still displayed.

Issue: The Customizer section for the previous page you were editing is still displayed. If you navigate back to the Pages panel, you see two pages listed. You're saying that you expect the previous section for a page to collapse when you navigate to another page in the preview? Or rather, you expect the Customizer section for the newly-navigated page to be automatically focused when navigating to a page?

**Comment by bsusala** [comment]
After reviewing the behaviour, neither :)

I am in favor of a more consistent UI. So:

CONTEXT
- editing in Customizer is, in my opinion, smth to be made after some minimal structure is in place, including static Pages
- using Customizer, in its early stage, doesn't require content editing, but more global settings
- when it's time to fine tune structure/content, navigating is more often done in Preview, rather than mingling in Customizer sections; so your Plugin should be Preview-centric, rather than Customizer-Sections-centric; all in all, your plugin is actually content focused, and content must be assessed in context (=Preview)
- links in Preview can lead to Pages, Posts or external resources, or CPTs, or other
- filtering them automatically in Customizer sections I think is possible (or not? I am not fluent enough to be certain)
- if I am in the Pages or Posts Section, I am expecting to Edit that - ie prior clicking on a type of object I am trying to edit is not relevant

SUGGESTION
- flush current categories, create a [NEW NAME TBD] Section for this plugin functionality
- whatever EDITABLE (supported) object is in Preview, will show the respective editing options, collapsed and ready to operate (except for Editor - its behaviour is absolutely ok and smart); context feedback will say "Editing [Page/Post/whatever]"
- resources lacking support for EDITING, yet, will show a message: "Not Editable, please do smth."
- when navigating away from current Preview:
-- if content modified, some message to Save/Quit
-- if nothing modified, close anything specific (Editor, section), refresh Section content according to new Object in Preview

Hope I wasn't rambling and be of some use with my ideas. I still have to debug why on my localdev, text from one page got saved on another page while navigating through Preview (initial issue). Will get back to that one I figure out.

**Comment by bsusala** [comment]
related: https://github.com/xwp/wp-customize-posts/issues/105#issue-150835037

**Comment by bsusala** [comment]
@jwold if I remember correctly, it was in HTML, since at the time, the plugin creates a standard TinyMCE instance. But since then, I wasn't able to reproduce the above initial possible bug behaviour.

**Comment by sayedwp** [comment]
@westonruter I checked, `editor.setContent` does not do anything when text tab is selected.  
https://github.com/xwp/wp-customize-posts/blob/develop/js/customize-post-section.js#L563
so the text just stays there, but doesn't  get updated in tinymce
This issue is very similar to the issue discussed [here](https://themeforest.net/forums/thread/wp-editor-tinymce-set-content-also-for-textarea-under-text-tab/100425), 
If I do `editor.save( { no_events: true } );`  after `editor.setContent` it works however the problem is that it wraps the text in p tag. Looking for a solution..

**Comment by westonruter** [comment]
Fixed in #198.
